pkg/healthcheck/healthcheck.go

```diff
@@ -749,7 +750,21 @@ func (hc *HealthChecker) checkNamespace(ctx context.Context, namespace string, s
 
 func (hc *HealthChecker) getDataPlanePods(ctx context.Context) ([]*pb.Pod, error) {
 	req := &pb.ListPodsRequest{}
-	if hc.DataPlaneNamespace != "" {
+	if hc.Options.TargetProxyResource != "" {
+		parts := strings.Split(hc.Options.TargetProxyResource, "/")
+		ownerType, ownerName := parts[0], parts[1]
+		ownerKind, err := k8s.CanonicalResourceNameFromFriendlyName(ownerType)
+		if err != nil {
+			return nil, err
```

Comment:
Hmm, I'm not exactly sure how to remedy this, but it's slightly problematic for this func to return a validation error to checks that are configured to retry. It means that the check will retry validation errors for up to 5 minutes, even if though they have no chance of succeeding. Maybe the best approach would be to add a new check prior to this check that just validates the target, and make that check non-retryable and fatal? That would cause the check command to exit immediately in the event of an invalid target, allowing the user to re-run the command with a valid target.

Reply:
well we should never hit this because we validate all this right after check command is processed here https://github.com/linkerd/linkerd2/pull/1777/files#diff-45365b61a96eb768eecd4c352cc736c1R143

Do you think that is not enough?

Reply:
Hmm, in that case, can we ensure that the `TargetProxyResource` field contains the canonical name, so that we don't have to call `CanonicalResourceNameFromFriendlyName` here, which could return an error?

Reply:
yeah, you're right. I'll do that